PowerNSX.psm1

```diff
@@ -22026,12 +22043,18 @@ function New-NsxFirewallSection  {
         [System.XML.XMLElement]$xmlRoot = $XMLDoc.CreateElement("section")
         $xmlDoc.appendChild($xmlRoot) | out-null
 
+        #Mandatory Fields
         Add-XmlElement -xmlRoot $xmlRoot -xmlElementName "name" -xmlElementText $Name
 
+        #Optional Fields
+        if ( $PsBoundParameters.ContainsKey('Universal')) {
```

Comment:
We need to test for $universal -eq $true (technically a user could pass -Universal:$false which would meet this condition incorrectly - which if they are wrapping this in a script and they are splatting parameters could easily happen.)  Also - you dont need to test if the user specified the switch as it has a default value (so you can just test it).  

if ( $Universal) {... would work fine here...